src/GraphQL/Utilities/Visitors/SchemaValidationVisitor.cs

```diff
@@ -197,6 +197,9 @@ public override void VisitInputObjectFieldDefinition(FieldType field, IInputObje
             {
                 throw new InvalidOperationException($"The default value of Input Object type field '{type.Name}.{field.Name}' is invalid.");
             }
+
+            if (field.Arguments?.Count > 0)
+                throw new InvalidOperationException($"The field '{field.Name}' of an Input Object type '{type.Name}' must not have any arguments specified.");
```

Comment:
New check. Relates to #1176 and linked issues about splitting input/output types. Now it will be better to have at least such a check.